### PR TITLE
Clean up timer resources properly when using RPCBus

### LINKED ISSUE #1062: Clean up timer resources in RPCBus
Memory profiling revealed to us that heavy usage of the RPCBus causes lots of lingering timers, due to the use of `time.After`.

### PR DESCRIPTION
**Pull request by julesdesmit**
Fixes #1062

### CODE REVIEW

File: pkg/util/nativeutils/rpcbus/rpcbus.go [hunk 1]
```diff
@@ -130,19 +130,31 @@ func (bus *RPCBus) Call(t topics.Topic, req Request, timeOut time.Duration) (int
...
+
 	select {
 	case reqChan <- req:
-	case <-time.After(timeOut):
+	case <-timer.C:
 		return bytes.Buffer{}, ErrRequestTimeout
 	}
 
+	if !timer.Stop() {
+		<-timer.C
```
Comment (goshawk-3) [inline pkg/util/nativeutils/rpcbus/rpcbus.go:142]: Should we drain this chan in a separate goroutine so we don't block `callTimeout`

**Comment by julesdesmit** [comment]
I don’t think that’s necessary, at most we are draining a single element.
If it becomes a performance bottleneck, we can always optimize later.

Op ma 31 mei 2021 om 19:20 schreef goshawk-3 ***@***.***>

File: pkg/util/nativeutils/rpcbus/rpcbus.go (hunk 1 above)
Comment (julesdesmit) [reply pkg/util/nativeutils/rpcbus/rpcbus.go:142]: I don’t think that’s necessary, at most we are draining a single element.
If it becomes a performance bottleneck, we can always optimize later.

Comment (goshawk-3) [reply pkg/util/nativeutils/rpcbus/rpcbus.go:142]: To be honest, I couldn't find where in `timerproc` or `deltimerLocked` the timer callback (the one that writes to timer.C) is called but let's rely on the example from documentation.

**Review by goshawk-3 — APPROVED** [review]

**Pull request merged**